CMakeLists.txt

```diff
@@ -96,39 +97,43 @@ set_option_if(ACTS_BUILD_PLUGIN_AUTODIFF ACTS_BUILD_EVERYTHING)
 include(GNUInstallDirs) # GNU-like installation paths, e.g. lib/, include/, ...
 include(ActsCompilerOptions) # default compile options
 include(ActsComponentsHelpers) # handle components via add_..._if commands
-include(TargetSourcesLocal) # needed only until we require CMake 3.13
 # place build products in `<build_dir>/bin` and `<build_dir>/lib` for simple use
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
 
+# dependency versions
+set(_acts_autodiff_version 0.5.11)
+set(_acts_boost_version 1.71)
+set(_acts_dd4hep_version 1.11)
+set(_acts_doxygen_version 1.8.15)
+set(_acts_eigen3_version 3.3.7)
+set(_acts_hepmc3_version 3.2.1)
+set(_acts_nlohmanjson_version 3.2.0)
+set(_acts_root_version 6.20)
+set(_acts_tbb_version 2020.1)
+
 # required packages
-# use only the CMake FindBoost module; avoids issues with the new cmake configs
-# that become available with Boost 1.70
-set(Boost_NO_BOOST_CMAKE ON)
```

Comment:
Is this not needed anymore?

Reply:
The variable is only used by the `FindBoost` CMake module shipped by CMake itself. Since the minimum required Boost version is 1.71, Boost always ships a proper CMake configuration file itself and the relevant `find_package(Boost ... CONFIG)` ensures that the old-style `FindBoost` module is not used.

Reply:
Hm. I remember the problem was that Sabrina wasn't able to build with Ubuntu and a **new** boost version that enabled this behavior. So basically the new behavior broke. Maybe this is fixed by the `CONFIG`.

Reply:
http://cmake.3232098.n2.nabble.com/Boost-1-70-0-FindBoost-issues-No-linker-libraries-found-td7599380.html This is the issue I was working around back then. I'm not sure, but maybe that was only in `1.70.0`, and is fixed in later versions.

Reply:
I suspect that this is fixed by using `CONFIG`, but I can not be exactly sure. If I remember correctly, Sabrina was using an old Ubuntu release which we do not support anymore.

Reply:
She was using an old Ubuntu release, but the problem was caused by using a newer boost version than was available out of the box.

Reply:
Can you check if you can reproduce the problem in boost `1.70.0` and goes away with `>1.70.0`? Otherwise I'd just keep the old behavior as long as we can.

Reply:
Yes, I will try to do that.

Reply:
I can not reproduce the problem with either `1.70` or `1.71`.

## Using `Boost_NO_BOOST_CMAKE` and **no** `CONFIG`:

### 1.70

```
-- Found Boost: /home/msmk/software/boost/v1.70.0/include (found suitable version "1.70.0", minimum required is "1.70") found components: filesystem program_options unit_test_framework
```

### 1.71

```
-- Found Boost: /home/msmk/software/boost/v1.71.0/include (found suitable version "1.71.0", minimum required is "1.70") found components: filesystem program_options unit_test_framework 
```
## Using `CONFIG` only

### 1.70

```
-- Found Boost 1.70.0 at /home/msmk/software/boost/v1.70.0/lib/cmake/Boost-1.70.0
--   Requested configuration: REQUIRED COMPONENTS filesystem;program_options;unit_test_framework
-- Found boost_headers 1.70.0 at /home/msmk/software/boost/v1.70.0/lib/cmake/boost_headers-1.70.0
-- Found boost_filesystem 1.70.0 at /home/msmk/software/boost/v1.70.0/lib/cmake/boost_filesystem-1.70.0
--   libboost_filesystem.a
-- Adding boost_filesystem dependencies: headers
-- Found boost_program_options 1.70.0 at /home/msmk/software/boost/v1.70.0/lib/cmake/boost_program_options-1.70.0
--   libboost_program_options.a
-- Adding boost_program_options dependencies: headers
-- Found boost_unit_test_framework 1.70.0 at /home/msmk/software/boost/v1.70.0/lib/cmake/boost_unit_test_framework-1.70.0
--   libboost_unit_test_framework.a
-- Adding boost_unit_test_framework dependencies: headers
```

## 1.71 (and manually setting `Boost_VERBOSE` to get the output):

```
-- Found Boost 1.71.0 at /home/msmk/software/boost/v1.71.0/lib/cmake/Boost-1.71.0
--   Requested configuration: REQUIRED COMPONENTS filesystem;program_options;unit_test_framework
-- Found boost_headers 1.71.0 at /home/msmk/software/boost/v1.71.0/lib/cmake/boost_headers-1.71.0
-- Found boost_filesystem 1.71.0 at /home/msmk/software/boost/v1.71.0/lib/cmake/boost_filesystem-1.71.0
--   [x] libboost_filesystem.so.1.71.0
--   [ ] libboost_filesystem.a
-- Adding boost_filesystem dependencies: headers
-- Found boost_program_options 1.71.0 at /home/msmk/software/boost/v1.71.0/lib/cmake/boost_program_options-1.71.0
--   [x] libboost_program_options.so.1.71.0
--   [ ] libboost_program_options.a
-- Adding boost_program_options dependencies: headers
-- Found boost_unit_test_framework 1.71.0 at /home/msmk/software/boost/v1.71.0/lib/cmake/boost_unit_test_framework-1.71.0
--   [x] libboost_unit_test_framework.so.1.71.0
--   [ ] libboost_unit_test_framework.a
-- Adding boost_unit_test_framework dependencies: headers
```

Reply:
Thanks. Then let's drop it and try to keep it in mind in case problem pop up again!